File: 02/solution.py
#!/usr/bin/env python

upperLimit = 4000000

fibIndexUpperLimit = 34


def fib(n, current = 0, next = 1):
  if n == 0:
    return current
  else:
    return fib(n-1, next, current+next)

print sum([fib(x) for x in xrange(34) if fib(x)%2 == 0 ])

File: 02/solution.go
package main

import (
  "fmt"
)

func main() {
  f := fib()

  total := 0
  for i := f(); i <= 4E6; i = f() {
    if i % 2 == 0 {
      total += i
    }
  }

  fmt.Println(total)
}

func fib() func() int {
  x,y := 1,1
  return func() int {
    x,y = y,x+y
    return x
  }
}
